lib/facter/facts/rhel/os/release.rb

```diff
@@ -27,7 +27,7 @@ def determine_release_version
           fact_value = {}
           fact_value['full'] = version
           fact_value['major'] = versions[0]
-          fact_value['minor'] = versions[1].gsub(/^0([1-9])/, '\1') if versions[1]
+          fact_value['minor'] = versions[1] if versions[1]
```

Comment:
Is there any test for the scenario that this change is fixing?

Reply:
no. from what we can tell at this moment, this code was copy-pasted from the debian os.release fact when the rhel fact was first written. it was not suppose to be there in the first place and it appears to be affecting only amazon 6.